src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -324,6 +324,11 @@ def forward(
             }
             key_states, value_states = past_key_value.update(key_states, value_states, self.layer_idx, cache_kwargs)
 
+        if attention_mask is not None:
+            seq_len = attention_mask.shape[1]
+            key_states = key_states[:, :, :seq_len]
+            value_states = value_states[:, :, :seq_len]
```

Comment:
ah, yeah static cache does not support that because it's input dependent, but great catch